esp/services/ws_workunits/ws_workunitsService.cpp

```diff
@@ -161,7 +164,6 @@ bool doAction(IEspContext& context, StringArray& wuids, CECLWUActions action, IP
                     cmd->getId(0,reply);
 
                     AuditSystemAccess(context.queryUserId(), true, "Updated %s", wuid);
-                    ensureWsWorkunitAccess(context, wuid, SecAccess_Write);
```

Comment:
Any idea why this check was performed down here instead of the beginning of this case ?

Reply:
It is because different action may need different permission.